app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -349,6 +340,61 @@ public function addColumn($columnId, $column)
         return $this;
     }
 
+    public function addColumnDefaultData(array $column): array
+    {
+        $config = $this->getConfigDefaultColumnSettings();
+        $columnHasIndex = array_key_exists('index', $column);
+        if ($columnHasIndex && array_key_exists($column['index'], $config['index'])) {
+            $column += $config['index'][$column['index']];
+        }
+
+        $columnHasType = array_key_exists('type', $column);
+        if ($columnHasType && array_key_exists($column['type'], $config['type'])) {
+            $column += $config['type'][$column['type']];
+        }
+
+        if ($columnHasType
+            && !array_key_exists('header', $column)
+            && in_array($column['type'], ['action', 'currency', 'price', 'store'])
+        ) {
+            switch ($column['type']) {
+                case 'action':
+                    $column['header'] = Mage::helper('adminhtml')->__('Action');
+                    break;
+                case 'currency':
+                case 'price':
+                    $column['header'] = Mage::helper('adminhtml')->__('Price');
+                    break;
+                case 'store':
+                    $column['header'] = Mage::helper('adminhtml')->__('Store View');
+                    break;
+            }
+        }
+
+        return $column;
+    }
+
+    public function getConfigDefaultColumnSettings(): array
+    {
+        if (is_null($this->defaultColumnSettings)) {
+            $configNode = Mage::getConfig()->getNode('grid/column/default');
+            # should be called only once
+            if ($configNode === false) {
+                Mage::app()->getCacheInstance()->cleanType('config');
```

Comment:
```suggestion
                Mage::app()->getCacheInstance()->cleanType('config');
                $configNode = Mage::getConfig()->getNode('grid/column/default');
```

Reply:
Clean cache is only required when you add this PR, but have config-cache enabled. In this case `getNode('grid/column/default')` will return false.

Reply:
Line 385: 
```php
$config = $configNode->asArray(); // $configNode can be false if not for assignment in line 384 
```

Reply:
Oh. Get the "Suggested change" wrong ... :)